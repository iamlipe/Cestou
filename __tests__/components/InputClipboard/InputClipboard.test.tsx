import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import InputClipBoard from '@/presentational/BasketScreen/InputClipboard';

describe('InputClipBoard', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<InputClipBoard text="test" />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<InputClipBoard text="test" />);
  });

  test('should render with text correctly', () => {
    const inputClipboard = renderWithThemeProvider(
      <InputClipBoard text="test" />,
    );

    const text = inputClipboard.queryByText('test');

    expect(text).toBeTruthy();
  });
});
