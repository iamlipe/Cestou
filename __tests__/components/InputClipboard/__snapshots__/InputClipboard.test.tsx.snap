// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputClipBoard should matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#00843F",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "justifyContent": "center",
        "marginBottom": 16,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 16,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      test
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï…‡
      </Text>
    </View>
  </View>
</View>
`;
